pkg/quota/v1/evaluator/core/pods.go

```diff
@@ -231,8 +231,9 @@ func enforcePodContainerConstraints(container *corev1.Container, requiredSet, mi
 	containerUsage := podComputeUsageHelper(requests, limits)
 	containerSet := quota.ToSet(quota.ResourceNames(containerUsage))
 	if !containerSet.Equal(requiredSet) {
-		difference := requiredSet.Difference(containerSet)
-		missingSet.Insert(difference.List()...)
+		if difference := requiredSet.Difference(containerSet); difference.Len() != 0 {
+			missingSet.Insert(fmt.Sprintf("%s: %s", container.Name, strings.Join(difference.List(), ",")))
```

Comment:
please rename missingSet to humanReadableMissingMessages.  I had to find the original messages to determine if it was human readable or not